src/block.c

```diff
@@ -343,8 +348,16 @@ struct block_device *block_device_duplicate(struct block_device *block)
 	if (block) {
 		result = calloc(1, sizeof(*result));
 		if (result) {
-			result->sysfs_path = str_dup(block->sysfs_path);
-			result->cntrl_path = str_dup(block->cntrl_path);
+			result->sysfs_path = strdup(block->sysfs_path);
+			result->cntrl_path = strdup(block->cntrl_path);
+
+			if (!result->sysfs_path || !result->cntrl_path) {
+				free(result->sysfs_path);
```

Comment:
```c
                        result->sysfs_path = strdup(block->sysfs_path);
			result->cntrl_path = strdup(block->cntrl_path);

			if (!result->sysfs_path || !result->cntrl_path) {
				free(result->sysfs_path);
				free(result->cntrl_path);
				free(result);
				return NULL;
			}
```

We dynamically allocate space for each string with `strdup`.  We are checking both and if either one is NULL, we free both.  Instead of checking for which one is non-null, we are letting free handle the `free(NULL)` case.

Reply:
Yup, thanks for clarification. Freeing NULL is not an issue.